File: clients/javascript/src/models/msgQueueReceiveIn.ts
// this file is @generated

export interface MsgQueueReceiveIn {
    topic: string;
    consumerGroup: string;
    batchSize?: number;
    leaseDurationMillis?: number;
}

export const MsgQueueReceiveInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MsgQueueReceiveIn {
        return {
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            batchSize: object['batch_size'],
            leaseDurationMillis: object['lease_duration_millis'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MsgQueueReceiveIn): any {
        return {
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'batch_size': self.batchSize,
            'lease_duration_millis': self.leaseDurationMillis,
        };
    }
}

File: clients/python/coyote/models/msg_queue_receive_in.py
# this file is @generated
import typing as t
from pydantic import Field

from ..internal.base_model import BaseModel


class MsgQueueReceiveIn(BaseModel):
    batch_size: t.Optional[int] = Field(default=None, alias="batch_size")

    lease_duration_millis: t.Optional[int] = Field(
        default=None, alias="lease_duration_millis"
    )


class _MsgQueueReceiveIn(BaseModel):
    topic: str

    consumer_group: str = Field(alias="consumer_group")

    batch_size: t.Optional[int] = Field(default=None, alias="batch_size")

    lease_duration_millis: t.Optional[int] = Field(
        default=None, alias="lease_duration_millis"
    )
